db/models/index.js

```diff
@@ -0,0 +1,83 @@
+const db = require('../index.js');
+const { aql } = require('arangojs');
+
+const MAX_RECORDS = 10000;
+const NUM_GAMES_PER_TRAIT = 4;
+
+/**
+ * Gets a list of traits for a product
+ * @param {String} id The ID of a product for which to find thje traits
+ * @returns {array} - An array of trait ids
+ */
+const fetchTraitsForProduct = async (id) => {
+  const edgesCollection = db.collection('edges');
+  id = 'games/' + String(id);
+
+  let results = await db.query(aql`
+        FOR doc in any ${id} ${edgesCollection}
+          OPTIONS { bfs: true, uniqueVertices: 'global' }
+          return { id: doc._id, name: doc.name }
+    `);
+
+  results = await results.all();
+  console.log(results);
+
+  return results;
+};
+
+/**
+ * Takes in a trait ID and gets four products with the same trait
+ * for practical reasons the list of games is limited to 10k results
+ * and four are randomly selected from that
+ * @param {string} id - ID of a trait for which to find some randomish games
+ */
+const fetchProductsForTrait = async (trait, game) => {
+  const edgesCollection = db.collection('edges');
+  //const whichGames = getRandomGames();
+  let count = 0;
+
+  let results = await db.query(aql`
+    FOR edge IN ${edgesCollection}
+      FILTER edge._to == ${trait}
+      LIMIT 10000
+      return edge._from
+  `);
+  results = await results.all();
+  results = getRandomGames(results);
+
+  return results;
+};
+
+const getTraitIdFromName = async (traitName) => {
+  const traitCollection = db.collection('traits');
+
+  try {
```

Comment:
Of the 3 function expressions in this file, this is the only one with a try...catch. Which is totally fine, of course! But your catch doesn't return out of the function.. I'm not in a position to check what would be returned from `getTraitIdFromName` when an error occurs, but I imagine that it would either return undefined, or not exit from the function at all, on account of it being a Promise? For either of those cases, it would be good practice to either handle errors here, or handle them in your `app.METHOD` lines in server.js. 

A suggestion for error handling: In `server.js`, for each endpoint:
```
app.get(ENDPOINT, async (req, res) => {
  // verify user input or parameters
  // declare globals needed in try & catch
  try {
    // all await statements here
  } catch (err) {
    console.error(err);
    res.status(500).send('An internal error occurred');
  }
});
```

And since your other two methods in `db/models/index.js` don't have try...catch blocks, `server.js` code that uses them definitely needs try...catch. Just a suggestion! If you have a better way of implementing error handling for those edge cases, by all means implement it the way you see fit.

Last note, an integration test suite for this edge case, and all other error edge cases related to using these methods would help you catch these edge cases before they make their way to a PR 👍

Reply:
HI Christina,

You're totally right about the lack of error handling. I was planning on adding it in when as I finish the full suite of crud functions. In this PR I was just focusing on getting the original code to work with the new database. I plan to tidy it up, get some tests written, and add error handling in the next PR.

Reply:
Cool cool! In that case count this as an "Approved" code review :)

Reply:
Awesome. Thanks for the feedback. I will apply it with the next PR.